### PR TITLE
[GenericPoseGraph] Add acceleration constraint.

### PR DESCRIPTION
**Pull request by pifon2a**

### CODE REVIEW

File: cartographer/pose_graph/constraint/cost_function/acceleration_cost_3d.h [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+            Map<const Eigen::Quaternion<T>>(imu_calibration) *
+            delta_velocity_imu_frame_.cast<T>() -
+        ((first_to_second_delta_time_seconds_ +
+          second_to_third_delta_time_seconds_) *
+         *gravity_constant / 2.f) *
+            Eigen::Vector3d::UnitZ().cast<T>();
+    const Matrix<T, 3, 1> start_velocity =
+        (Map<const Matrix<T, 3, 1>>(middle_position) -
+         Map<const Matrix<T, 3, 1>>(start_position)) /
+        T(first_to_second_delta_time_seconds_);
```
Comment (mbokeloh) [inline cartographer/pose_graph/constraint/cost_function/acceleration_cost_3d.h]: Shouldn't those be static_cast<T>()? Also below.

File: cartographer/pose_graph/constraint/cost_function/acceleration_cost_3d_test.cc [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+using testing::ParseProto;
+
+using PositionType = std::array<double, 3>;
+using RotationType = std::array<double, 4>;
+using ResidualType = std::array<double, 3>;
+
+constexpr char kParameters[] = R"PROTO(
+  delta_velocity_imu_frame { x: 1 y: 1 z: 1 }
+  first_to_second_delta_time_seconds: 10.0
+  second_to_third_delta_time_seconds: 10.0
```
Comment (mbokeloh) [inline cartographer/pose_graph/constraint/cost_function/acceleration_cost_3d_test.cc]: Nit: Use a different value here to make the test tighter.

**Comment by pifon2a** [comment]
@wally-the-cartographer merge

**Comment by wally-the-cartographer** [comment]
Merge requested by authorized user pifon2a. Merge queue now has a length of 1.

**Comment by wally-the-cartographer** [comment]
Refusing to merge, since the combined checks state is failure

**Pull request merged**